plugins/ccgx/fu-ccgx-dock-bb.c

```diff
@@ -0,0 +1,734 @@
+		guint64 tmp = fu_common_strtoull (value);
+		if (tmp < G_MAXUINT16) {
+			self->quirks_silicon_id = tmp;
+			return TRUE;
+		}
+		g_set_error_literal (error,
+				     G_IO_ERROR,
+				     G_IO_ERROR_INVALID_DATA,
+				     "invalid DeviceSiliconID");
+		return FALSE;
+	}
+	if (g_strcmp0 (key, "DeviceFWAppType") == 0) {
+		guint64 tmp = fu_common_strtoull (value);
+		if (tmp < G_MAXUINT16) {
+			self->quirks_fw_app_type = tmp;
+			return TRUE;
+		}
+		g_set_error_literal (error,
+				     G_IO_ERROR,
+				     G_IO_ERROR_INVALID_DATA,
+				     "invalid DeviceFWAppType");
+		return FALSE;
+	}
+	if (g_strcmp0 (key, "UpdateMessage") == 0) {
+		self->fw_update_message = g_strdup (value);
+		return TRUE;
+	}
+	if (g_strcmp0 (key, "UpdateMessagePrimary") == 0) {
+		self->fw_update_message_primary = g_strdup (value);
+		return TRUE;
+	}
+	if (g_strcmp0 (key, "UpdateMessageBackup") == 0) {
+		self->fw_update_message_backup = g_strdup (value);
+		return TRUE;
+	}
+	if (g_strcmp0 (key, "ModelName") == 0) {
+		self->model_name = g_strdup (value);
+		return TRUE;
+	}
+	return FALSE;
+}
+
+static gboolean
+fu_ccgx_dock_bb_probe (FuDevice *device, GError **error)
+{
+	FuCcgxDockBb *self = FU_CCGX_DOCK_BB (device);
+	g_autofree gchar *devid_bb = NULL;
+
+	/* get vid and pid of device */
+	devid_bb = g_strdup_printf ("USB\\VID_%04X&PID_%04X&bb", 
+				   (guint) fu_usb_device_get_vid (FU_USB_DEVICE (device)),
+				   (guint) fu_usb_device_get_pid (FU_USB_DEVICE (device)));
+
+	/* this will trigger setting up all the quirks */
+	fu_device_add_instance_id (device, devid_bb);
+
+	self->dm_device = DM_DEVICE_NONE;
+
+	/* set device type and fw image type according to custom flag in quirk data */
+	if (fu_device_has_custom_flag (FU_DEVICE(device),"cy-device-external-bb")) {
+		self->dm_device = DM_DEVICE_EXTERNAL_BB;
+		self->usb_inf_num = USB_HID_INF_NUM;
+	} else if (fu_device_has_custom_flag (FU_DEVICE(device),"cy-device-pd-i2c")) {
+		self->dm_device = DM_DEVICE_PD_I2C;
+		self->usb_inf_num = USB_I2C_INF_NUM;
+	} else {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "not supported device");
+		return FALSE;
+	}
+
+	self->fw_image_type = FW_IMAGE_TYPE_NONE;
+	if (self->dm_device  == DM_DEVICE_PD_I2C) {
+		if (fu_device_has_custom_flag (FU_DEVICE(device),"cy-fw-image-single")) {
+			self->fw_image_type = FW_IMAGE_TYPE_SINGLE;
+		} else if (fu_device_has_custom_flag (FU_DEVICE(device),"cy-fw-image-dual-symmetric")) {
+			self->fw_image_type = FW_IMAGE_TYPE_DUAL_SYMMETRIC;
+		} else if (fu_device_has_custom_flag (FU_DEVICE(device),"cy-fw-image-dual-asymmetric")) {
+			self->fw_image_type = FW_IMAGE_TYPE_DUAL_ASYMMETRIC;
+		} else {
+			g_set_error(error, FWUPD_ERROR,
+				    FWUPD_ERROR_NOT_SUPPORTED,
+				    "not supported fw image");
+			return FALSE;
+		}
+		if (fu_device_has_custom_flag (FU_DEVICE(device),"cy-fw-primary-update-only"))
+			self->fw_primary_update_only = TRUE;
```

Comment:
This seems unused, right?

Reply:
Yes, It is optional. You can remove this flag.